developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -487,6 +487,7 @@ notice that `key` is mandatory, since we will present it to the user when select
 * `class` - additional html classes to put on the `<th>` element
 * `attr_callbacks` - a hash where the key is the name of html attribute, and the value is a function in the form `->(host) { attribute_value }`
 * `callback` - a function that receives the host model and returns html content for the `<th>` element: `->(host) { "<span>...</span>".html_safe }`
+* `export_key` - a string that is used to derive exported column's name and value from. Passing `reported_data.sockets` results into header called `Reported Data - Sockets` and value corresponding to the result of calling `host&.reported_data&.sockets`. Alternatively, for cases where WebUI columns aggregate multiple values, this can be an array of strings to split the values into their own columns in the export.
```

Comment:
I guess we need to find a better way than `export_key: ['string', 'string']`. Currently, `export_key: ['content_facet_attributes.errata_counts', 'content_facet_attributes.upgradable_rpm_count']` will add to csv
```csv
Content Facet Attributes - Errata Counts,Content Facet Attributes - Upgradable Rpm Count
"{:security=>0, :bugfix=>0, :enhancement=>0, :total=>0, :applicable=>{:security=>0, :bugfix=>0, :enhancement=>0, :total=>0}}",0
```

Which seems OK if a simple info retrieval can be made (like with RPM count), but it's not useful in case there is a hash that needs parsing. Also, `Content Facet Attributes` is unnecessary long and descriptive label, especially if it's repeated. I get, that CSV is not meant for people's eyes, but still can be converted to e.g. table.

I thought about `export_data:` which could be used to parse complicated cases, and `export_key:` for either simple cases or simply labeling. This would look like repeating of `callback` logic, but reduced to return simple data. Not sure if we want that for probably the only corner case though...

Reply:
Things were getting somewhat unwieldy with some things being string/symbol/array of either/proc and trying to combine multiple sets of those into something that would make sense. I didn't want to spend too much time considering it is a bit of an edge case.

I added export_data key, the value is expected to be an instance of newly introduced class ExportDefinition. If export_data is provided, all information is pulled from it, shadowing export_key. Is this good enough?

Reply:
Yeap, tested with that PR, seems OK to me.